lib/scanner/walk.go

```diff
@@ -476,8 +479,11 @@ func SymlinkTypeEqual(disk, index uint32) bool {
 	// know means you are on Unix, and on Unix you don't really care what the
 	// target type is. The moment you do know, and if something doesn't match,
 	// that will propagate through the cluster.
-	if disk&protocol.FlagSymlinkMissingTarget != 0 && index&protocol.FlagSymlinkMissingTarget == 0 {
+	if disk == symlinks.TargetUnknown {
```

Comment:
Yeah, this is what I mentioned in the description on the PR. The comment basically says that if we don't know what type it is, we should consider it equal for now. This what the new code does, but not the old...